### PR TITLE
add TemperaturechangerDevice

### PR DESCRIPTION
**Pull request by michelececcacci**
Not really sure about this, @smagnaniuni can you take a quick look?

### CODE REVIEW

**Comment by smagnaniuni** [comment]
Seems ok as a start.

A few things:
- There is `protected double STEP;` unused, probably for future use but I don't understand what is it for.

- For the state value:
I don't understand what the current value is: temperature, delta temperature, intensity?

Linked to this, due to #16 and #19, I was thinking of some things I wanted to implement:
- For the currentValue:
Due to multiple devices interfacing with the current temperature (#16), and to the single responsibility principle, a HomeEnvironment class should be created (#19).
This will also contain other environmental variables such as the air quality if it's ok for you, but I think it makes sense like this.

- For the temperature:
We need to implement a data type for the temperature, since a double alone does not explain much (is it Celsius, Kelvin,..?)
This allows the class to deal with the conversion stuff and would allow new methods to be added in the future (eg. getKelvin()).

```mermaid
classDiagram
    class Temperature {
        +double getCelsius()
        +double getFahrenheit()
    }

    class CelsiusTemperature {
        -double degCelsius
    }

    Temperature <|-- CelsiusTemperature
```

**Comment by michelececcacci** [comment]
Yeah, i should have pointed it out. `STEP` is just a variable that should be used to tweak the function (and maybe to make conversions easier). Current value is the temperature. I absolutely support creating a homeEnvironment variable. I will also start working on the temperature class right now, if that's alright with you.

**Comment by smagnaniuni** [comment]
I don't think it should return the temperature but a delta temperature / intensity.
The device can be connected to the environment object and they should talk that way.
But this doesn't matter a lot, aside from the simulation fidelity. If you return the temperature value, the controller will have to communicate to the environment the new one.
The current value would be communicated from the controller, which in turns is given by the temperature sensor.

Please open an issue and assign yourself if you are creating the temperature type.

I will implement the home environment and temperature sensor as soon as the temperature type is added.

**Review by aleemont1 — COMMENTED** [review]
LGTM. Please comment your code before asking for reviews, even if it's very simple.

**Comment by michelececcacci** [comment]
@smagnaniuni  some feedback on this would be nice. I think it's fine, and i'm adding tests rn.

File: app/src/main/java/homer/api/DeviceIdImpl.java [hunk 1]
```diff
@@ -9,10 +9,7 @@ public final class DeviceIdImpl implements DeviceId {
 
     private final UUID id = UUID.randomUUID();
 
-    /**
-     * 
-     * @return The device unique id.
-     */
+    @Override
```
Comment (smagnaniuni) [inline app/src/main/java/homer/api/DeviceIdImpl.java:12]: Re-commit causes conflict

File: app/src/test/java/homer/model/temperaturechangers/HeaterTest.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package homer.model.temperaturechangers;
+
+public class HeaterTest {
```
Comment (smagnaniuni) [inline app/src/test/java/homer/model/temperaturechangers/HeaterTest.java:3]:
```suggestion
class HeaterTest {
```

File: app/src/test/java/homer/model/temperaturechangers/ACTest.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package homer.model.temperaturechangers;
+
+
+/**
+ * Tests AC. still WIP
+ */
+public class ACTest {
```
Comment (smagnaniuni) [inline app/src/test/java/homer/model/temperaturechangers/ACTest.java:7]:
```suggestion
class ACTest {
```

File: app/src/main/java/homer/temperaturechangers/AC.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+public class AC extends AbstractTemperatureChanger implements TemperatureChanger {
+
+    public AC(double minIntensity, double maxIntensity, double currentIntensity, Environment environment) {
+        super(minIntensity, maxIntensity, currentIntensity, environment);
+    }
+
+    @Override
+    public void updateTick(Duration deltaTime) {
+        var oldTemp = this.environment.getTemperature().getCelsius();
+        var newTemp = Math.max(this.minTemperature.getCelsius(), oldTemp - this.intensity * deltaTime.toMillis() * this.STEP);
```
Comment (smagnaniuni) [inline app/src/main/java/homer/temperaturechangers/AC.java:17]: These newTemp = Math.min / max are not immediately understandable at first sight

**Review by smagnaniuni — COMMENTED** [review]
Missing finals in various places + see comments

File: app/src/main/java/homer/temperaturechangers/AC.java (hunk 4 above)
Comment (michelececcacci) [reply app/src/main/java/homer/temperaturechangers/AC.java:17]: what do you suggest here? using a new variable for the possible new max? renaming variables?

**Pull request closed without merging**